autonomic-plugin-common/src/main/java/br/com/autonomiccs/autonomic/plugin/common/guru/AutonomiccsSystemVmsGuru.java

```diff
@@ -53,48 +53,73 @@
 import br.com.autonomiccs.autonomic.plugin.common.beans.AutonomiccsSystemVm;
 import br.com.autonomiccs.autonomic.plugin.common.daos.AutonomiccsSystemVmDao;
 
+/**
+ * This class provides methods for the deployment and management of system virtual machines used by
+ * the Autonomiccs platform.
+ */
 @Component
 public class AutonomiccsSystemVmsGuru implements VirtualMachineGuru, InitializingBean {
 
     private final Logger logger = LoggerFactory.getLogger(getClass());
 
     @Autowired
-    private String ipAddressInternet;
+    protected String ipAddressInternet;
     @Autowired
-    private String maskForIpAddressInternet;
+    protected String maskForIpAddressInternet;
 
     @Autowired
-    private VirtualMachineManager virtualMachineManager;
+    protected VirtualMachineManager virtualMachineManager;
 
     @Autowired
-    private AutonomiccsSystemVmDao autonomiccsSystemVmDao;
+    protected AutonomiccsSystemVmDao autonomiccsSystemVmDao;
 
     @Autowired
-    private ConfigurationDao configurationDao;
+    protected ConfigurationDao configurationDao;
 
     @Autowired
-    private DataCenterDao datacenterDao;
+    protected DataCenterDao datacenterDao;
 
+    /**
+     * It inicializes the Autonomiccs System Vms Guru.
+     */
     @Override
     public void afterPropertiesSet() throws Exception {
         logger.debug("Autonomiccs system VMs guru initialized.");
         virtualMachineManager.registerGuru(VirtualMachine.Type.Instance, this);
     }
 
     /**
-     * For now we do not do anything here, shit design
+     * For now we do not do anything here. Unfortunately we have to stick with this design to follow
+     * the Apache CloudStack flow.
      */
     @Override
     public boolean finalizeCommandsOnStart(Commands arg0, VirtualMachineProfile arg1) {
         return true;
     }
 
+    /**
+     * It finalizes the Autonomiccs virtual machine deployment. First it gets the
+     * {@link AutonomiccsSystemVm} with the profile id; get the destination data center; configure
+     * the VM network cards; and updates the 'AutonomiccsSystemVm' table with the deployed system
+     * vm.
+     */
     @Override
     public boolean finalizeDeployment(Commands cmds, VirtualMachineProfile profile, DeployDestination dest, ReservationContext context) throws ResourceUnavailableException {
         AutonomiccsSystemVm autonomiccsSystemVm = autonomiccsSystemVmDao.findById(profile.getId());
 
         DataCenter dc = dest.getDataCenter();
         List<NicProfile> nics = profile.getNics();
+
+        setIpAddressOfEachNic(autonomiccsSystemVm, dc, nics);
+
+        autonomiccsSystemVmDao.update(profile.getId(), autonomiccsSystemVm);
+        return true;
+    }
+
+    /**
+     * It configures the each network interface card IP address of the {@link AutonomiccsSystemVm}.
```

Comment:
It configures each network interface card IP address of the {@link AutonomiccsSystemVm}.